src/main/java/ru/testtask/service/UserService.java

```diff
@@ -0,0 +1,47 @@
+package ru.testtask.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import ru.testtask.model.User;
+import ru.testtask.repo.UserRepo;
+
+import java.util.Optional;
+
+@Service
+public class UserService implements UserDetailsService {
+
+    @Autowired
+    private UserRepo userRepo;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userRepo.findByUsername(username);
+
+        if (user == null){
+            throw new UsernameNotFoundException("User does not exists");
+        }
+
+        return user;
+    }
+
+    public User getUserByUsername(String username){
+        return userRepo.findByUsername(username);
+    }
+
+    public String getCurrentUserId(){
+        org.springframework.security.core.userdetails.User user = (org.springframework.security.core.userdetails.User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
The method can be used from thread that doesn't know about current context. We need to check that context is available. 
Please call following from controller:
```
new Thread(() -> {
            org.springframework.security.core.userdetails.User user = (org.springframework.security.core.userdetails.User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
        }).start();
```